packages/world/src/modules/keysintable/KeysInTableModule.sol

```diff
@@ -0,0 +1,77 @@
+import { ResourceSelector } from "../../ResourceSelector.sol";
+
+import { KeysInTableHook } from "./KeysInTableHook.sol";
+import { KeysInTable, KeysInTableTableId } from "./tables/KeysInTable.sol";
+import { UsedKeysIndex, UsedKeysIndexTableId } from "./tables/UsedKeysIndex.sol";
+
+/**
+ * This module deploys a hook that is called when a value is set in the `sourceTableId`
+ * provided in the install methods arguments. The hook keeps track of the keys that are in a given table.
+ * This mapping is stored in a table registered by the module at the `targetTableId` provided in the
+ * install methods arguments.
+ *
+ * Note: if a table with composite keys is used, only the first key is indexed
+ *
+ * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
+ * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
+ */
+contract KeysInTableModule is IModule, WorldContext {
+  using ResourceSelector for bytes32;
+
+  // The KeysInTableHook is deployed once and infers the target table id
+  // from the source table id (passed as argument to the hook methods)
+  KeysInTableHook immutable hook = new KeysInTableHook();
+
+  function getName() public pure returns (bytes16) {
+    return bytes16("index");
+  }
+
+  function install(bytes memory args) public override {
+    // Extract source table id from args
+    bytes32 sourceTableId = abi.decode(args, (bytes32));
+
+    IBaseWorld world = IBaseWorld(_world());
+
+    bytes32 resourceSelector = ResourceSelector.from(KeysInTableTableId.getNamespace(), KeysInTableTableId.getName());
+
+    if (ResourceType.get(resourceSelector) == Resource.NONE) {
+      // Register the tables
+      world.registerTable(
+        KeysInTableTableId.getNamespace(),
+        KeysInTableTableId.getName(),
+        KeysInTable.getSchema(),
+        KeysInTable.getKeySchema()
+      );
+      world.registerTable(
+        UsedKeysIndexTableId.getNamespace(),
+        UsedKeysIndexTableId.getName(),
+        UsedKeysIndex.getSchema(),
+        UsedKeysIndex.getKeySchema()
+      );
+
+      // Register metadata for the tables
+      (string memory tableName1, string[] memory fieldNames1) = KeysInTable.getMetadata();
+      world.setMetadata(KeysInTableTableId.getNamespace(), KeysInTableTableId.getName(), tableName1, fieldNames1);
+      (string memory tableName2, string[] memory fieldNames2) = UsedKeysIndex.getMetadata();
+      world.setMetadata(UsedKeysIndexTableId.getNamespace(), UsedKeysIndexTableId.getName(), tableName2, fieldNames2);
+
+      // Grant the hook access to the tables
+      world.grantAccess(KeysInTableTableId.getNamespace(), KeysInTableTableId.getName(), address(hook));
+      world.grantAccess(UsedKeysIndexTableId.getNamespace(), UsedKeysIndexTableId.getName(), address(hook));
+    }
+
+    // Register a hook that is called when a value is set in the source table
+    StoreSwitch.registerStoreHook(sourceTableId, hook);
```

Comment:
tiny nit, but since we call the other methods above on `world`, I'd call `registerStoreHook` on `world` as well for consistency